### DISCUSSION TITLE
How to filter records by join table

### DISCUSSION THREAD

**Discussion by Triquetra** [emmett-framework/emmett#341] [Q&A] [answered]
Given the following Models:

```python
class User(Model):
    name = Field()
    has_many('memberships', {'organizations': {'via': 'memberships'}})


class Organization(Model):
    name = Field()
    has_many('memberships', {'users': {'via': 'memberships'}})


class Membership(Model):
    belongs_to('user', 'organization')
    role = Field()
```

How would one get a list of User records with membership at Organization X and role == 'admin'?

The documentations talks about how to add a record with a role on the join table, but I can't figure out how to access the join table data from the User records.

**Comment by gi0baro** [comment, marked as answer]
@Triquetra you can reach the same result in different ways.

In case you have the organization record:

```python
org = Organization.get(name="X")
admin_users = org.users.switch(Membership).where(lambda m: m.role == "admin").select()
```

or without `switch`:

```python
admin_users = org.users.where(Membership.role == "admin").select()
```

with a single query using `switch`:

```python
admin_users = User.all().join('organizations').switch(Organization).where(
    lambda o: o.name == "X"
).switch(Membership).where(
    lambda m: m.role == "admin"
).select()
```

with a single explicit query:

```python
User.all().join('organizations').where(
    (Organization.name == "X") & (Membership.role == "admin")
).select()
```

**Reply by Triquetra** [reply]
I cannot find any documentation on the switch method. Could you point me to the appropriate page so that I can learn more about it?

**Reply by gi0baro** [reply]
@Triquetra It seems I forgot to document it. Will open up an issue to update documentation.

**Reply by Triquetra** [reply]
This was very instructive. Thank you, @gi0baro!
